components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/model/PingFederateDCRClient.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.pingfederate.model;
+
+import feign.Headers;
+import feign.Param;
+import feign.RequestLine;
+import feign.Response;
+
+/**
+ * DCR API Client.
+ */
+public interface PingFederateDCRClient {
+
+    @RequestLine("POST")
+    @Headers("Content-Type: application/json")
+    public Response createApplication(ClientInfoList clientInfo);
+
+    @RequestLine("GET /{clientId}")
+    @Headers("Content-Type: application/json")
+    public ClientInfo getApplication(@Param("clientId") String clientId);
+
+    @RequestLine("PUT ")
```

Comment:
Don't we have to provide the {clientId} for update

Reply:
no pings rest api was not in that mode